react/components/atoms/container-swipeable/container-swipeable.js

```diff
@@ -12,81 +24,156 @@ export class ContainerSwipeable extends PureComponent {
         };
 
+        this._initialViewPositionY = ScreenHeight;
         this.animating = false;
+
+        this.panResponder = PanResponder.create({
+            onStartShouldSetPanResponder: (_evt, _gestureState) => true,
+            onPanResponderGrant: this.onPanResponderGrant,
+            onPanResponderMove: this.onPanResponderMove,
+            onPanResponderRelease: this.onPanResponderRelease
+        });
     }
 
+    onPanResponderGrant = (_evt, _gestureState) => {
+        this._initialViewPositionY = this.state.containerInnerPositionAnimated.__getValue().y;
+    };
+
+    onPanResponderMove = (_evt, gestureState) => {
+        if (gestureState.dy === 0) {
+            return;
+        }
+
+        //if the user is going up, we slow down the up movement  on each interaction
+        let gestureStateDistanceY = gestureState.dy > 0 ? gestureState.dy : gestureState.dy / 2;
+
+        const nextViewPositionY = this._nextViewPositionY(gestureStateDistanceY);
+        const hasReachedMaxTopPosition = nextViewPositionY <= 0;
+
+        if (hasReachedMaxTopPosition) {
+            return;
+        }
+
+        this.state.containerInnerPositionAnimated.setValue({
+            x: 0,
+            y: nextViewPositionY
+        });
+    };
+
+    onPanResponderRelease = (_evt, gestureState) => {
+        const nextViewPositionY = this._nextViewPositionY(gestureState.dy);
+        const isInsideOfCloseThreshold =
+            nextViewPositionY - this.props.hideThreshold >
+            ScreenHeight - this.state.containerChildrenHeight;
+
+        if (isInsideOfCloseThreshold) {
+            this.toggle();
+        } else {
+            const toValue = { x: 0, y: this._initialViewPositionY };
+
+            Animated.spring(this.state.containerInnerPositionAnimated, {
+                toValue,
+                duration: this.props.animationsDuration
+            }).start();
+        }
+    };
+
     toggle = () => {
         if (this.animating) {
             return;
         }
 
-        if (this.state.visible) {
+        if (this.state.itemsVisible) {
             this.setState(
                 {
-                    visible: false
+                    itemsVisible: false
                 },
-                () => {
-                    this._toggleAnimations();
-                }
+                this._toggleAnimations
             );
         } else {
-            this.setState(
-                {
-                    visible: true,
-                    modalVisible: true
-                },
-                () => {
-                    this._toggleAnimations();
-                }
-            );
+            this.setState({
+                itemsVisible: true,
+                modalVisible: true
+            });
         }
     };
 
     onLayout = event => {
-        this.setState({
-            containerChildrenHeight: event.nativeEvent.layout.height
-        });
+        if (this.state.containerChildrenHeight) {

```

Comment:
Update to oneliner for better readability ? :)
`if (this.state.containerChildrenHeight) return;`